pkg/provisioner/ironic/ready_test.go

```diff
@@ -0,0 +1,144 @@
+package ironic
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/metal3-io/baremetal-operator/pkg/bmc"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/clients"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/testserver"
+	"github.com/stretchr/testify/assert"
+)
+
+func buildServer(t *testing.T, name string, v1 string, drivers string) *testserver.MockServer {
+	return testserver.New(t, name).Response("/v1", v1).
+		Response("/v1/drivers", drivers)
+}
```

Comment:
IMHO I preferred the previous pattern (based on Fluent Builder and Object Mother), since it was aiming to improve the readability in the test cases definition. I found a little bit misleading have a single helper method with parameters instead of a fluent interface. I think that now that we have a nice middleware helper object (`testserver`) now a fluent interface could be built easily on top of it (with maybe distinct types for distinct servers, to avoid any confusion)

Reply:
I'm not sure exactly what you mean here. This function cut down on some rather long lines that repeated a few times in the test setup. Do you mean replace `buildServer` with something like `ironicServer(t, drivers)` and `inspectorServer(t)`? Or something else?

Reply:
Added a PR to show better the idea: please see https://github.com/dhellmann/baremetal-operator/pull/1

Reply:
I've incorporated those changes here. As we write more tests using that approach, I feel like we're going to end up with more and more types based on `MockServer`. I'm not sure how I feel about that. It seems like it could be positive, because we'll have little server types that are used only in part of the test suite. On the other hand, there will be a lot of them.

Reply:
Thanks. My general expectation is to enrich the fluent interface - when required - of the defined mock types to add other behaviors rather than making new types.